get-started/nodejs/build-images.md

```diff
@@ -0,0 +1,224 @@
+
+`class MyImage extends NodeBaseImage {}`
+
+This would create a class called `MyImage` that inherited functionality from the base class `NodeBaseImage`.
+
+In the same way, when we use the `FROM` command, we tell Docker to include in our image all the functionality from the `node:12.18.1` image.
+
+>**Note**
+>
+> If you want to learn more about creating your own base images, see [Creating base images](https://docs.docker.com/develop/develop-images/baseimages/).
+
+To make things easier when running the rest of our commands, let’s create a working directory. This instructs Docker to use this path as the default location for all subsequent commands. This way we do not have to type out full file paths but can use relative paths based on the working directory.
+
+`WORKDIR /app`
+
+Usually the very first thing you do once you’ve downloaded a project written in Node.js is to install npm packages. This will ensure that your application has all its dependencies installed into the `node_modules` directory where the Node runtime will be able to find them.
+
+Before we can run `npm install`, we need to get our `package.json` and `package-lock.json` files into our images. We use the `COPY` command to do this. The  `COPY` command takes two parameters. The first parameter tells Docker what file(s) you would like to copy into the image. The second parameter tells Docker where you want that file(s) to be copied to. We’ll copy the `package.json` and `package-lock.json` file into our working directory `/app`.
+
+```
+COPY package.json package.json
+COPY package-lock.json package-lock.json
+```
+
+Once we have our `package.json` files inside the image, we can use the `RUN` command to execute the command npm install. This works exactly the same as if we were running npm install locally on our machine, but this time these Node modules will be installed into the `node_modules` directory inside our image.
+
+`RUN npm install`
+
+At this point, we have an image that is based on node version 12.18.1 and we have installed our dependencies. The next thing we need to do is to add our source code into the image. We’ll use the COPY command just like we did with our `package.json` files above.
+
+`COPY . .`
+
+The COPY command takes all the files located in the current directory and copies them into the image. Now, all we have to do is to tell Docker what command we want to run when our image is run inside of a container. We do this with the `CMD` command.
+
+`CMD [ "node", "server.js" ]`
+
+Here's the complete Dockerfile.
+
+```
+FROM node:12.18.1
+
```

Comment:
Maybe also add an environment

```dockerfile
ENV NODE_ENV=production
```

This also tells the Node.js interpreter to run  in production mode. (https://expressjs.com/en/advanced/best-practice-performance.html#set-node_env-to-production)

Just as a good practice.